web-extension/engine/bookmycomics.js

```diff
@@ -248,9 +251,9 @@ BmcEngine.prototype.register = function(label, cb) {
     return this._db.registerComic(label, this._comic.reader, this._comic.info, err => {
         LOGS.debug('S11', {'err': err});
         if (!err) {
-            this._forceMemoizeComic();
+            this._forceMemoizeComic(() => cb(err));
+            return ;
         }
-        return cb(err);
```

Comment:
beware, you're no longer calling the callback if `this._db.registerComic()` encountered an error.